### PR TITLE
Fix mkdocs relpath

### PR DESCRIPTION
**Pull request by stsewd**
After https://github.com/rtfd/readthedocs.org/pull/5377/
The code calling the config module is in charge of getting the full
path.

### CODE REVIEW

**Comment by davidfischer** [comment]
I encountered this error [here](https://readthedocs.org/projects/rtd-mkdocs-test-project/builds/9154099/). I was able to reproduce it on the same repo locally. This PR did fix the issue.

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 1]
```diff
@@ -248,7 +248,7 @@ def build(self):
             '--site-dir',
             self.build_dir,
             '--config-file',
-            self.yaml_file,
+            os.path.relpath(self.yaml_file, self.root_path),
```
Comment (stsewd) [inline readthedocs/doc_builder/backends/mkdocs.py]: One test was expecting this call to be relative to the checkout path, but with the new change this was a full path.
Relative path is shorter to show in the output and we are already setting the cwd to the checkout path, so we are fine.

**Comment by stsewd** [comment]
I also checked all the usages of `self.yaml_file`, they are good

**Comment by ericholscher** [comment]
@stsewd do you think this should be hotfixed today?

**Comment by stsewd** [comment]
@ericholscher yeah, it affects people using mkdocs with a rtd configuration file. Which I think are most of our users now

**Pull request merged**